apollo-api/src/commonMain/kotlin/com/apollographql/apollo/api/internal/MapResponseParser.kt

```diff
@@ -0,0 +1,88 @@
+package com.apollographql.apollo.api.internal
+
+import com.apollographql.apollo.api.CustomScalarAdapters
+import com.apollographql.apollo.api.Error
+import com.apollographql.apollo.api.Operation
+import com.apollographql.apollo.api.Response
+import com.apollographql.apollo.api.Response.Companion.builder
+import com.apollographql.apollo.api.ResponseField
+import com.apollographql.apollo.api.parseData
+
+/**
+ * [MapResponseParser] parses network responses, including data, errors and extensions from a regular Map<String, Any?>.
+ * For better performance, you can parse a response from a [okio.BufferedSource] directly with [StreamResponseParser].
+ * That will avoid the cost of having to create an entire Map in memory
+ */
+object MapResponseParser {
```

Comment:
just a thought should we call it BufferedResponseParser?

Reply:
I like having `Map` vs `Stream`. Buffer is more ambiguous as technically the `StreamResponseParser` works on a `BufferedSource`. At least `Map` is nowhere near `StreamResponseParser`.